tests/dragon/test_environment_loader.py

```diff
@@ -93,59 +97,70 @@ def test_environment_loader_FLI_fails(monkeypatch):
 
 
 @pytest.mark.parametrize(
-    "expected_keys, expected_values",
+    "feature_stores",
     [
-        pytest.param(["key1", "key2", "key3"], ["value1", "value2", "value3"]),
-        pytest.param(["another key"], ["another value"]),
+        pytest.param([], id="No feature stores"),
+        pytest.param([MemoryFeatureStore()], id="Single feature store"),
+        pytest.param(
+            [MemoryFeatureStore(), FileSystemFeatureStore()],
+            id="Multiple feature stores",
+        ),
     ],
 )
-def test_environment_loader_memory_featurestore(
-    expected_keys, expected_values, monkeypatch
+def test_environment_loader_featurestores(
+    feature_stores: t.List[FeatureStore], monkeypatch: pytest.MonkeyPatch
 ):
-    """MemoryFeatureStores can be correctly serialized and deserialized"""
-    feature_store = MemoryFeatureStore()
-    key_value_pairs = zip(expected_keys, expected_values)
-    for k, v in key_value_pairs:
-        feature_store[k] = v
-    monkeypatch.setenv(
-        "SSFeatureStore", base64.b64encode(pickle.dumps(feature_store)).decode("utf-8")
-    )
-    config = EnvironmentConfigLoader()
-    config_feature_store = config.get_feature_store()
+    """FeatureStore can be correctly identified, serialized and deserialized"""
+    with monkeypatch.context() as m:
+        for fs in feature_stores:
+            value = base64.b64encode(pickle.dumps(fs)).decode("utf-8")
+            key = f"SSFeatureStore.{fs.descriptor}"
+            m.setenv(key, value)
+
+        config = EnvironmentConfigLoader()
+        actual_feature_stores = config.get_feature_stores()
+
+        for fs in feature_stores:
+            # Confirm that the descriptors were used as keys in the loaded feature stores
+            assert fs.descriptor in actual_feature_stores
 
-    for k, _ in key_value_pairs:
-        assert config_feature_store[k] == feature_store[k]
+            # Confirm that the value loaded from env var is a FeatureStore
+            # and it is consistent w/the key identifying it
+            loaded_fs = actual_feature_stores[fs.descriptor]
+            assert loaded_fs.descriptor == fs.descriptor
 
 
 @pytest.mark.parametrize(
-    "expected_keys, expected_values",
+    "value_to_use,error_filter",
```

Comment:
More of a general comment, but shouldn't there be a test with a `DragonFeatureStore`?